Public/Src/Cache/ContentStore/UtilitiesCore/FileSystemConstants.cs

```diff
@@ -12,6 +13,26 @@ namespace BuildXL.Cache.ContentStore.Interfaces.FileSystem
     /// </summary>
     public static class FileSystemConstants
     {
+        private static class LibC
+        {
+            /// <nodoc />
+            [DllImport("libc", SetLastError = true)]
+            public static extern int chmod(string path, int mode);
+        }
+
+        /// <nodoc />
+        public static void OverrideFileAccessMode(bool changePermissions, string path)
+        {
+#if PLATFORM_OSX
+            if (changePermissions)
+            {
+                // Force 0777 on the file at 'path' - this is a temporary hack when placing files as our cache layer
+                // currently does not track Unix file access flags when putting / placing files
+                LibC.chmod(path, 0x0001 | 0x0002 | 0x0004 | 0x0040 | 0x0080 | 0x0100 | 0x0008 | 0x0010 | 0x0020);

```

Comment:
This looks clearer to me:
```suggestion
                LibC.chmod(path, Convert.ToInt32("777", 8));
```

Reply:
👍